pattern_language/core-language/data-types.md

```diff
@@ -147,11 +147,26 @@ Sometimes arrays need to keep on growing as long as a certain condition is met.
 u8 string[while(std::mem::read_unsigned($, 1) != 0xFF)] @ 0x00;
 ```
 
+For more complex conditions, combine an array to `eof` with a struct containing `if` statements. Using `continue` would move on to the next array item of the loop; using `break` would stop the loop and the growing of the array. See also the [pattern control flow](control-flow.md#pattern-control-flow).
```

Comment:
```suggestion
A more complex example: growing an array until `eof` with a struct containing `if` statements. Using `continue` would move on to the next array item of the loop; using `break` would stop the loop and the growing of the array. See also the [pattern control flow](control-flow.md#pattern-control-flow).
```

Reply:
I am not sure what you are referring to by "Using `continue` would move on to the next array item of the loop; using `break` would stop the loop and the growing of the array."

If I am not mistaken, the following syntax: `Command command_list[while(!std::mem::eof())] @ 0x00;` doesn't permit using `break` or `continue` keywords since you can't specify the loop body.